packages/calypso-e2e/src/lib/utils/editor-tracks-event-manager.ts

```diff
@@ -0,0 +1,119 @@
+import { Locator, Page } from 'playwright';
+import { TracksEvent, TracksEventProperties } from '../../types';
+
+// Exporting just for unit testing
+export const createEventMatchingPredicate = (
+	expectedName: string,
+	matchingProperties?: TracksEventProperties
+) => {
+	return ( event: TracksEvent ) => {
+		const [ name, properties ] = event;
+		if ( expectedName !== name ) {
+			return false;
+		}
+
+		if ( ! matchingProperties ) {
+			return true; // Nothing more to check, we're done here!
+		}
+
+		for ( const expectedProperty in matchingProperties ) {
+			if ( properties[ expectedProperty ] !== matchingProperties[ expectedProperty ] ) {
+				return false;
+			}
+		}
+
+		return true;
+	};
+};
+
+const sleep = async ( ms: number ) => new Promise( ( resolve ) => setTimeout( resolve, ms ) );
+
+/**
+ * A class wrapping the editor Tracks events monitored in the browser
+ */
+export class EditorTracksEventManager {
+	private page: Page;
+	private editor: Locator;
+
+	/**
+	 * Construct an instance of the editor Tracks event manager
+	 *
+	 * @param page The Playwright page
+	 */
+	constructor( page: Page ) {
+		this.page = page;
+
+		// The Tracks events are stashed on "window" in whatever iframe is the actual editor (whether post, page, or site).
+		// We need to consolidate down to a single locator that works across all editors, and handles if we have the Gutenframe wrapper.
+		// It just matters that we're in the right iframe, so we use a very generic selector ("body") to get something safe and top-level.
+		if ( this.page.url().includes( '/wp-admin' ) ) {
```

Comment:
I don't have a problem with this approach, though a word of caution is that when the Atomic editor is loaded (with third party cookie disabled) the page does not immediately navigate away to the non-framed editor. 

This means the URL will not contain `wp-admin` until the iframe load is rejected. This was one of the reasons I removed this logic from the Atomic refactor.

Reply:
Yeah, I had tried this approach where we just searched multiple `iframes` for what looked like the right window, but it ended up not being super clean because our need to use the `evaluate` function, and because I don't want to make any assumptions about caching a JSHandle pointer to `window`.

But you bring up a great point -- this definitely should _not_ be in the constructor! Who knows what the URL might be then? The good news is these methods will only be called once the editor is fully loaded, but we need to make the handling here happen in real time! I will update that.